src/records/RecHttp.cc

```diff
@@ -859,60 +865,58 @@ convert_alpn_to_wire_format(std::string_view protocols, unsigned char *wire_form
 
   // Parse the comma separated protocol string into a list of protocol names.
   std::vector<std::string_view> alpn_protocols;
-  std::string_view protocol;
-  size_t pos                  = 0;
+  TextView protocol;
   int computed_alpn_array_len = 0;
-  while (pos < protocols.size()) {
-    size_t next_pos = protocols.find(',', pos);
-    if (next_pos == std::string_view::npos) {
-      protocol = protocols.substr(pos);
-      pos      = protocols.size();
-    } else {
-      protocol = protocols.substr(pos, next_pos - pos);
-      pos      = next_pos + 1;
-    }
+
+  // Make sure there are no whitespace characters in the protocols string.
+  if (std::any_of(protocols.begin(), protocols.end(), [](char const c) { return isspace(c); })) {
+    Error("The configured ALPN string contains whitespace characters: %.*s", static_cast<int>(protocols.size()), protocols.data());
```

Comment:
Current implementation is more accurate actually. Strictly speaking, a protocol id can have spaces and also commas.

Reply:
> Current implementation is more accurate actually. Strictly speaking, a protocol id can have spaces and also commas.

True, but the ones we support do not.

This comes from an internal review comment from @ywkaras . The advantage of this check is that it will give an explicit error message to someone who might try this:

```
CONFIG proxy.config.ssl.client.alpn_protocols STRING "h2, http/1.1"
```

Note the space after the comma. Without the space check from this patch, the user will get a generic `Unknown protocol name in configured ALPN list:  http/1.1`. This Error message will explicitly point out that a space is the issue.

Reply:
I hope we won't need to support ids with spaces. This is one of reasons why I didn't want to use protocol ids.

It's easy to prohibit spaces, but if users might try that format, wouldn't it be more user friendly if we ignore spaces?

Reply:
Yeah, that's a good point. Let me try with trimming. TextView has some nice APIs to make that easy.

Reply:
I updated the parsing logic to allow the protocols to be padded with white space.